### PR TITLE
Downsample fanout

### PR DESCRIPTION
**Pull request by derek-pryor**
Support for new downsample_volume Step Function. Updated to lambda build process.

Linked to
* https://github.com/jhuapl-boss/boss-tools/pull/2
* https://github.com/jhuapl-boss/boss/pull/8
* https://github.com/jhuapl-boss/spdb/pull/1

### CODE REVIEW

File: salt_stack/salt/build-lambda/files/makedomainenv [hunk 1]
```diff
@@ -0,0 +1,118 @@
+#! /usr/bin/env bash
```
Comment (movestill) [inline salt_stack/salt/build-lambda/files/makedomainenv]: Something went very wrong with the merge. This is the pre-Python 3.6 lambda file.

File: salt_stack/salt/build-lambda/init.sls [hunk 2]
```diff
@@ -0,0 +1,71 @@
+# Setup the lambda development/build server.
```
Comment (movestill) [inline salt_stack/salt/build-lambda/init.sls]: This looks like it was based on an old Salt formula? I updated `salt_stack/salt/lambda-dev/init.sls` for Python 3.6.

File: salt_stack/salt/python/python36.sls [hunk 3]
```diff
@@ -0,0 +1,49 @@
+# Build Python 3.6.x from source and install to /usr/local/bin.
```
Comment (movestill) [inline salt_stack/salt/python/python36.sls]: I'm package installing Python 3.6 on the build server (also in `salt_stack/salt/lambda-dev/init.sls`).

File: salt_stack/salt/top.sls [hunk 4]
```diff
@@ -27,6 +27,9 @@ base:
     'lambda*':
```
Comment (movestill) [inline salt_stack/salt/top.sls]: This now builds the Python 3.6 lambda build server.

File: salt_stack/salt/build-lambda/files/makedomainenv (hunk 1 above)
Comment (derek-pryor) [reply salt_stack/salt/build-lambda/files/makedomainenv]: Hm, I wonder if some of my initial work with getting downsample to work in lambda messed up the merge somehow.

File: .gitmodules [hunk 5]
```diff
@@ -1,28 +1,28 @@
 [submodule "salt_stack/salt/boss-tools/files/boss-tools.git"]
 	path = salt_stack/salt/boss-tools/files/boss-tools.git
 	url = ../boss-tools.git
-	branch = integration	
+	branch = downsample-fanout
```
[the remark this answers is not in the document]
Comment (derek-pryor) [reply .gitmodules]: Reverted to integration so there is no problem with the merge

**Pull request merged**